internal/provider/resource_account.go

```diff
@@ -163,7 +163,7 @@ func resourceAccountRead(ctx context.Context, d *schema.ResourceData, meta inter
 
 	arr, err := aClient.Read(ctx, d.Id())
 	if err != nil {
-		if apiErr := api.AsServerError(err); apiErr != nil && apiErr.Status == int32(http.StatusNotFound) {
+		if apiErr := api.AsServerError(err); apiErr != nil && apiErr.Kind == codes.NotFound.String() {
```

Comment:
For programmatic checking like this one I trust int comparisons more than string comparisons since changes in casing or typos are easier to mess things up.  Here and everywhere else in this PR.

```suggestion
		if apiErr := api.AsServerError(err); apiErr != nil && apiErr.ResponseStatus() == http.StatusNotFound {
```